### ISSUE TITLE
Cannot use Enter key to select from POS list in Firefox

### ISSUE THREAD

**Issue by flathat** [CORE-POS/IS4C#452]
This, or something like it, seems to come up from time to time so I'll establish this issue to collect observations about it.  The issue was dealt with in #318 for a change in Firefox 24.

I reported in the Friday conference call today that I could not select from a member list using Enter, or leave it with CL-Enter, in Firefox 35.0.1. Two others on the call couldn't replicate. I closed the window that POS session was in and opened POS in a tab of another Firefox window that was already open and there select with Enter works as it should.  Not the first time the cannot-select-by-Enter has happened for me, both in Firefox and Chrome, but the first time I've just tried it again in a different window. Select-by-Enter is fine in Chrome 40.0.2214.111 at the moment.

**Comment by gohanman** [comment]
This is what I would do for basic debugging:
040f72913ea2b4e1ef9778c56e41b8871ee9e2a6

I happen to like the Firebug plugin when working in Firefox although the built-in developer console is likely fine. I'm sure Chrome has something similar. Either way the console.log call should write messages to the developer console.

What I'm trying to establish here is:
# Is the keydown event handler running?
# What keycode is keydown receiving?
# Is the keyup event handler running?
# What keycode is the keyup handler receiving?
# What is the sequence of key events?

Potential problems:
* One or both key event handlers is not running at all
  * Solution(s): make sure select has input focus. Check for jquery errors. Try different versions of jquery.
* One or both key event handlers is not running on the enter key
  * Solution(s): try both enter keys if there's a numpad enter and an enter above shift. Check programming if it's a programmable keyboard. Check language / keyboard layout settings maybe?
* Enter key is sending a keycode other than 13
  * Solution(s): figure out why this is happening; add support for an alternate enter keycode if appropriate
* A key event is occurring *between* the enter keydown event and the enter keyup event
  * Solution(s): figure out why this is happening; the keycode of the inbetween event should make it fairly obvious

Background:
The javascript key event handling is in place rather than simply letting enter submit the form for two reasons. First, to make the clear button work, it's tracking the two keystrokes before enter so that the special sequence C+L+enter cancels the operation. Second, it's trying to ensure the form only submits on an enter keyup immediately preceeded by an enter keydown. It's possible depending on computer speed, browser behavior, keyboard springiness, and who knows what else a key event can get split across two pages. The cashier press enter down which causes the a form to submit. While the enter key is still in the "down" state the list page loads. When the enter key releases, the list page receives the keyup event. This event should *not* immediately submit the form.

I do not believe this relates to the older issue circa Firefox 23. At that time CORE was using a keypress event handler instead of keyup or keydown. Newer versions of Firefox do not raise a keypress event on the enter key.

**Comment by flathat** [comment]
In Firefox the JS log can be displayed with Ctrl-Shift-J.
Tools > Web Developer > Web Console 
displays the same but as one tab in a larger toolset.
I haven't been able to find it in Firebug.

I made Andy's change to `$LANE/js/selectSubmit.js` and followed it in the console.
At some point something changed and the log said:
> The Web Console logging API (console.log, console.info, console.warn, console.error) has been disabled by a script on this page.

after which `console.log()` output didn't appear.
The member list also stopped responding to `Enter` around that time, I'm not sure which came first.
Closing the browser, not just the tab, and re-opening brought both the response to `Enter` and `console.log()` back.
